pystac_client/client.py

```diff
@@ -236,45 +299,62 @@ def from_dict(
         return result
 
     @lru_cache()
-    def get_collection(self, collection_id: str) -> Optional[Collection]:
+    def get_collection(
+        self, collection_id: str
+    ) -> Optional[Union[Collection, CollectionClient]]:
         """Get a single collection from this Catalog/API
 
         Args:
             collection_id: The Collection ID to get
 
         Returns:
-            CollectionClient: A STAC Collection
+            Union[Collection, CollectionClient]: A STAC Collection
         """
-        if self._supports_collections() and self._stac_io:
-            url = self._get_collections_href(collection_id)
+        collection: Union[Collection, CollectionClient]
+
+        if self.conforms_to(ConformanceClasses.COLLECTIONS) or self.conforms_to(
+            ConformanceClasses.FEATURES
+        ):
+            assert self._stac_io is not None
+
+            url = self._collections_href()
+            url = f"{url.rstrip('/')}/{collection_id}"
```

Comment:
```suggestion
            url = self._collection_href(collection_id)
```

This feels like something that could/should be its own private function, in case we need to do this logic again later.

Reply:
yeah maybe. It used to be baked into the `_get_collections_href` helper. I didn't like the branch in there to support one use, but I can put it back. I definitely don't feel strongly about that.